scripts/backup_restore_simulation.sh

```
@@ -0,0 +1,34 @@
+docker compose exec -T postgres pg_dump -U file_engine file_engine > "$DB_DUMP"
+
+echo "[backup] step 2/4 storage snapshot -> $STORAGE_TAR"
+BASE_ROOT="${FILE_BASE_ROOT:-.data/file-engine}"
+mkdir -p "$BASE_ROOT"
+tar -czf "$STORAGE_TAR" -C "$BASE_ROOT" .
+
+echo "[restore] step 3/4 replay db dump"
+cat "$DB_DUMP" | docker compose exec -T postgres psql -U file_engine file_engine >/dev/null
+
+echo "[restore] step 4/4 replay storage snapshot"
+rm -rf "$BASE_ROOT"/*
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to /* .

(SC2115)